test/common/event/dispatcher_impl_test.cc

```diff
@@ -339,6 +340,228 @@ TEST(TimerImplTest, TimerEnabledDisabled) {
   EXPECT_FALSE(timer->enabled());
 }
 
+// Timers scheduled at different times execute in order.
+TEST(TimerImplTest, TimerOrdering) {
+  Api::ApiPtr api = Api::createApiForTest();
+  DispatcherPtr dispatcher(api->allocateDispatcher("test_thread"));
+
+  ReadyWatcher watcher1;
+  Event::TimerPtr timer1 = dispatcher->createTimer([&] { watcher1.ready(); });
+
+  ReadyWatcher watcher2;
+  Event::TimerPtr timer2 = dispatcher->createTimer([&] { watcher2.ready(); });
+
+  ReadyWatcher watcher3;
+  Event::TimerPtr timer3 = dispatcher->createTimer([&] { watcher3.ready(); });
+
+  timer1->enableTimer(std::chrono::milliseconds(0));
+  timer2->enableTimer(std::chrono::milliseconds(1));
+  timer3->enableTimer(std::chrono::milliseconds(2));
+
+  // Sleep for 5ms so timers above all trigger in the same loop iteration.
+  usleep(5000);
```

Comment:
Do we have anything in the test framework? I think this will choke windows build.

Reply:
Good call.  I found the absl sleep functions which are used in some places in the test framework.